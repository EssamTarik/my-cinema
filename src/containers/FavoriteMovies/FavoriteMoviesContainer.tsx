import React from "react";
import { IProps } from "./interfaces";
import MovieGridContainer from "../MovieGrid";
import { IMovie } from "../../store/movieSearch/models";

const FavoriteMoviesContainer = ({ favorites }: IProps) => {
  const favoriteMovies = Object.values(favorites) as IMovie[];
  return <MovieGridContainer movies={favoriteMovies} />;
};

export default FavoriteMoviesContainer;
