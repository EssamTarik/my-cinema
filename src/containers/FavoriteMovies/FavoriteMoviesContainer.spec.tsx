import React from "react";
import { render, cleanup } from "@testing-library/react";

import FavoriteMoviesContainer from "./FavoriteMoviesContainer";

jest.mock("../MovieGrid", () => () => <div />);
const mockMovie = jest.requireMock("movie");

describe("FavoriteMoviesContainer", () => {
  afterEach(cleanup);
  it("renders", () => {
    const { asFragment } = render(
      <FavoriteMoviesContainer favorites={{ [mockMovie.id]: mockMovie }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
