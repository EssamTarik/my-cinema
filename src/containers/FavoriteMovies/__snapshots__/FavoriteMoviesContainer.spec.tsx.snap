// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteMoviesContainer renders 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
